### DISCUSSION TITLE
Issues with zoom and "allow reload" functionality in MacOS SEB 3.0 config file preferences

### DISCUSSION THREAD

**Discussion by sk-bt** [SafeExamBrowser/seb-mac#154] [Q&A] [answered]
Am using the new MacOS SEB 3.0 for a test that we will be giving next Spring ('22) and ran into some issues with the zoom and "allow reload" functionalities. Has anyone encountered this/know of a solution?

Problem: **Even though "Allow reload" is disabled for both the Exam Window/Additional Windows, both "Cmd + R" and using the reload command from the "View" drop-down on the menu bar still allow the page to be reloaded.** This is problematic for our test since our testing system will automatically boot a student out of the test and require a resetting process by a testing administrator if the page is reloaded in the middle of their exam. 

Additionally, **both page and text zoom are enabled and the "zoom in/out" commands can be clicked on in the "View" drop-down, but neither the keyboard shortcut or "View" drop-down actually zoom in/out on the browser page.** The OS recognizes that the keys are being pressed to zoom in/out but not actually doing the action. That being said, if I enable the "browser window toolbar" in "User Interface", the buttons that appear for increasing/decreasing page and text zooms work fine, it's just the keyboard shortcuts and View menu that are non-functional even though they appear normal. 

Tested this on a Macbook Pro and the issue was replicated in Big Sur (11.6.1), Catalina (don't remember exact version) and Mojave as well. I cleared out all other SEB files/remnants (deleted files, issued the "defaults delete org.safe.....") so this is not a case of another config file accidentally being used. 

These behaviors happen on other websites outside of my test page's URL (I tried on ESPN.com and same issues occurred) so it appears to be the app rather than the specific web page. 

I also returned to the config file and SEB version (2.2.2) we used to test last Spring (2021) that uses the same URL and configuration options and found the zoom in/out functionality works perfectly while reloading is disabled as expected. 

Worst comes to worst, we can always re-package the 2.2.2 version of the application but I was wondering if I'm missing something new in the config file that would be overriding these functionalities or if it's an issue with the 3.0 build itself. Let me know if you would like video/photos of the issue or any of my other configuration settings for further context and I can post them ASAP.

**Comment by danschlet** [comment, marked as answer]
Thanks for reporting this issue, the keyboard shortcuts somehow got broken completely in the major refactoring of the browser engine in SEB 3.0. We will fix than in the SEB 3.1 update, which ideally should be available before the end of this year. See this issue: https://github.com/SafeExamBrowser/seb-mac/issues/157

**Reply by sk-bt** [reply]
Thank you for the quick response and additional info! Good to know about the development timeline and when to expect a possible fix.

**Reply by danschlet** [reply]
I fixed the issues you reported in the [SEB 3.2pre4 pre-release version](https://github.com/SafeExamBrowser/seb-mac/releases/tag/3.2pre4). Besides what you found out, also back/forward navigation using cmd-Cursor left/right was possible. Would be appreciated if you could test it.

SEB 3.2 only runs on macOS 10.14 Mojave or newer, I will back port these fixes to SEB 3.1 though.
